drools-parser/src/main/java/org/drools/parser/DRLVisitorImpl.java

```diff
@@ -86,23 +88,47 @@ public Object visitLhs(DRLParser.LhsContext ctx) {
 
     @Override
     public Object visitLhsPatternBind(DRLParser.LhsPatternBindContext ctx) {
-        // TODO: this logic will likely be split to visitLhsPattern
         if (ctx.lhsPattern().size() == 1) {
-            DRLParser.LhsPatternContext lhsPatternCtx = ctx.lhsPattern(0);
-            currentPattern = new PatternDescr(lhsPatternCtx.objectType.getText());
+            Object result = super.visitLhsPatternBind(ctx);
```

Comment:
HI @tkobayas 
I'm trying to see the `super` method but I can not find "DRLParserBaseVisitor" class: where it should be ?
I would like to see if it would be possible to have something more defined then just "Object" here, not sure if it is possible....

Reply:
Hi @gitgabrio ,
`DRLParserBaseVisitor` will be generated when you run `mvn clean install`.

```
./target/generated-sources/antlr4/org/drools/parser/DRLParserBaseVisitor.java
```

I see your point of "something more than `Object`". At the moment, we use the return value only for `String` constraint (so related methods like `visitExpression`, `visitLiteral`). In other visitor methods like `visitLhsPattern`, it actually returns null. The goal of `DRLVisitorImpl` is to produce `PackageDescr` and it is conveniently accessible as an instance field.

So... probably we have not yet fully make use of the return object but we may find some more usage as we will implement the visitor further and then we may revisit the appropriate type for the return object.

Reply:
@tkobayas  Thanks for explanation! :+1: